packages/web-components/src/components/va-modal/va-modal.tsx

```diff
@@ -0,0 +1,336 @@
+    const initialFocus = (this.el.querySelector(this.initialFocusSelector) ||
+      this.el.shadowRoot.querySelector(
+        this.initialFocusSelector,
+      )) as HTMLElement;
+    this.focusTrap = createFocusTrap(
+      [this.el, this.alertActions, this.closeButtonContainer],
+      {
+        // trap is removed in teardownModal - disable escape key deactivating the focus trap
+        escapeDeactivates: false,
+        // the element we want to target immediately after opening modal
+        initialFocus: initialFocus || this.closeButton,
+      },
+    );
+    this.focusTrap.activate();
+    disableBodyScroll(this.el);
+    this.undoAriaHidden = hideOthers(this.el);
+
+    // NOTE: With this PR (https://github.com/department-of-veterans-affairs/vets-website/pull/11712)
+    // we rely on the existence of `body.modal-open` to determine if a modal is
+    // currently open and adjust programmatic scrolling if there is.
+    document.body.classList.add('modal-open');
+
+    // Conditionally track the event.
+    if (!this.disableAnalytics) {
+      const detail = {
+        componentName: 'va-modal',
+        action: 'show',
+        details: {
+          status: this.status,
+          title: this.modalTitle,
+          primaryButtonText: this.primaryButtonText,
+          secondayButtonText: this.secondaryButtonText,
+        },
+      };
+      this.componentLibraryAnalytics.emit(detail);
+    }
+  }
+
+  /**
+   * This method removes the focus trap, re-enables scrolling and
+   * removes aria-hidden="true" from external elements.
+   */
+  private teardownModal() {
+    this.focusTrap.deactivate();
+    clearAllBodyScrollLocks();
+    this.undoAriaHidden();
+
+    document.body.classList.remove('modal-open');
+  }
+
+  render() {
+    const {
+      modalTitle,
+      hideCloseButton,
+      primaryButtonClick,
+      primaryButtonText,
+      secondaryButtonClick,
+      secondaryButtonText,
+      status,
+      visible,
+    } = this;
+
+    if (!visible) return null;
+
+    const ariaLabel = `${modalTitle} modal` || 'Modal';
+    const wrapperClass = status
+      ? 'va-modal-inner va-modal-alert'
+      : 'va-modal-inner';
+    const bodyClass = status ? 'va-modal-alert-body' : 'va-modal-body';
+    const titleClass = status ? 'va-modal-alert-title' : 'va-modal-title';
+    const contentClass = status ? 'va-modal-alert-text' : undefined;
+    const ariaRole = status => {
+      if (status === 'warning' || status === 'error') {
+        return 'alertdialog';
+      }
+      return 'dialog';
+    };
+    const btnAriaLabel = modalTitle
+      ? `Close ${modalTitle} modal`
+      : 'Close modal';
+
+    return (
+      <Host
+        aria-label={ariaLabel}
+        aria-modal="true"
+        class="va-modal"
+        role={ariaRole(status)}
+      >
+        <div class={wrapperClass} tabIndex={-1}>
+          {!hideCloseButton && (
+            <div ref={el => (this.closeButtonContainer = el as HTMLDivElement)}>
```

Comment:
Do we need this container? I'm not sure what it's doing, and it's not interactive so I don't see us focusing it.

Reply:
Good catch. I'll remove this.

Reply:
We need this container for alt-tabbing to work properly. I tried setting the focus trap to `[this.el, this.alertActions, this.wrapper]` and that only works for tabbing. I'm assuming that when we try to alt-tab out of the light DOM it focuses the last focusable element of `this.wrapper` instead of the behavior that we want which is to focus the close button and complete the loop.